src/middleware/devtools.ts

```diff
@@ -17,326 +18,201 @@ type DevtoolsType = {
   error: (payload: any) => void
 }
 
-export type NamedSet<T extends State> = {
-  <
-    K1 extends keyof T,
-    K2 extends keyof T = K1,
-    K3 extends keyof T = K2,
-    K4 extends keyof T = K3
-  >(
-    partial: PartialState<T, K1, K2, K3, K4>,
-    replace?: boolean,
-    name?: string | { type: unknown }
-  ): void
+type Devtools = <
+  T extends State,
+  Mps extends [StoreMutatorIdentifier, unknown][] = [],
+  Mcs extends [StoreMutatorIdentifier, unknown][] = [],
+  U = Partial<T>
+>(
+  initializer: StateCreator<T, [...Mps, ['zustand/devtools', unknown]], Mcs>,
+  options?: DevtoolsOptions
+) => StateCreator<T, Mps, [['zustand/devtools', U], ...Mcs]>
+
+declare module '../vanilla' {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
```

Comment:
I think this is a good rule as it also applies to runtime unused variables. This case is a rare one so I think it's okay to suppress the warning instead of removing the rule entirely.